__tests__/schema/opportunity.ts

```diff
@@ -2174,3 +2186,156 @@ describe('mutation editOpportunity', () => {
+        placeholder
+        opportunityId
+      }
+    }
+  `;
+
+  it('should require authentication', async () => {
+    await testMutationErrorCode(
+      client,
+      {
+        mutation: MUTATION,
+        variables: { id: opportunitiesFixture[0].id },
+      },
+      'UNAUTHENTICATED',
+    );
+  });
+
+  it('should throw error when user is not a recruiter for opportunity', async () => {
+    loggedUser = '2';
+
+    await testMutationErrorCode(
+      client,
+      {
+        mutation: MUTATION,
+        variables: {
+          id: opportunitiesFixture[0].id,
+        },
+      },
+      'FORBIDDEN',
+      'Access denied!',
+    );
+  });
+
+  it('should throw error when opportunity already has questions', async () => {
+    loggedUser = '1'; // user 1 is recruiter for opportunitiesFixture[0]
+
+    await testMutationErrorCode(
+      client,
+      {
+        mutation: MUTATION,
+        variables: { id: opportunitiesFixture[0].id },
+      },
+      'CONFLICT',
+    );
+  });
+
+  it('should recommend and persist screening questions', async () => {
+    loggedUser = '1';
+
+    const opportunity = await con.getRepository(OpportunityJob).save(
+      con.getRepository(OpportunityJob).create({
+        title: 'Opportunity without questions',
+        tldr: 'TLDR',
+        state: OpportunityState.DRAFT,
+        location: [
+          {
+            type: LocationType.HYBRID,
+            city: 'Varaždin',
+            subdivision: 'Varaždinska',
+            country: 'Croatia',
+          },
+        ],
+        meta: {
+          seniorityLevel: SeniorityLevel.SENIOR,
+          employmentType: EmploymentType.PART_TIME,
+        },
+        content: {
+          requirements: { content: 'Requirements', html: '' },
+          responsibilities: { content: 'Responsibilities', html: '' },
+        },
+      }),
+    );
+
+    await con.getRepository(OpportunityUser).save({
+      opportunityId: opportunity.id,
+      userId: '1',
+      type: OpportunityUserType.Recruiter,
+    });
+
+    let clientSpy: jest.SpyInstance | undefined;
+
+    jest
+      .spyOn(gondulModule, 'getGondulClient')
+      .mockImplementationOnce((): ServiceClient<typeof GondulService> => {
+        const transport = createMockGondulTransport();
+        const client = createClient(GondulService, transport);
+
+        clientSpy = jest.spyOn(client, 'screeningQuestions');
+
+        return {
+          instance: client,
+          garmr: createGarmrMock(),
+        };
+      });
+
+    const res = await client.mutate(MUTATION, {
+      variables: { id: opportunity.id },
+    });
+
+    expect(res.errors).toBeFalsy();
+
+    expect(clientSpy).toHaveBeenCalledTimes(1);
+    expect(clientSpy).toHaveBeenCalledWith({
+      jobOpportunity: `**Location:** HYBRID, Varaždin, Varaždinska, Croatia
+**Job Type:** PART_TIME
+**Seniority Level:** SENIOR
+
+### Overview ###
+TLDR
+
+### Responsibilities ###
+Responsibilities
+
+### Requirements ###
+Requirements`,
```

Comment:
@davidercruz this is the data I have currently, made it as close to your example from here https://github.com/dailydotdev/bragi/pull/107